From 94bf3cba3df8cdbc1bae805bf878bb8f0743e317 Mon Sep 17 00:00:00 2001
From: Mike Pagano <mpagano@gentoo.org>
Date: Thu, 11 Oct 2012 20:45:26 -0400
Subject: [PATCH] tg3: Apply short DMA frag workaround to 5906 

Backport of b7abee6ef888117f92db370620ebf116a38e3f4d for 3.0.X

tg3: Apply short DMA frag workaround to 5906

5906 devices also need the short DMA fragment workaround.  This patch makes the necessary change.

Signed-off-by: Mike Pagano <mpagano@gentoo.org>
---
 drivers/net/tg3.c |    5 +++--
 1 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/drivers/net/tg3.c b/drivers/net/tg3.c
index c4ab8a7..45a4cbd 100644
--- a/drivers/net/tg3.c
+++ b/drivers/net/tg3.c
@@ -13685,8 +13685,9 @@ static int __devinit tg3_get_invariants(struct tg3 *tp)
 	 */
 	tg3_flag_set(tp, 4G_DMA_BNDRY_BUG);
 
-	if (tg3_flag(tp, 5755_PLUS))
-		tg3_flag_set(tp, SHORT_DMA_BUG);
+	if (tg3_flag(tp, 5755_PLUS) ||
+		GET_ASIC_REV(tp->pci_chip_rev_id) == ASIC_REV_5906)
+			tg3_flag_set(tp, SHORT_DMA_BUG);
 	else
 		tg3_flag_set(tp, 40BIT_DMA_LIMIT_BUG);
 
-- 
1.7.8.6
